src/lib/utils/shex.js

```diff
@@ -311,27 +312,32 @@ const validateList = async (paths, shape) => {
    */
   window.fetch = solid.fetch;
 
-  const validPaths = [];
+  let validQuads = [];
 
   // Load the notification shape using the ShEx Loader
-  await Loader.load([shape], [], paths, []).then(loaded => {
+  const loaded = await Loader.load([shape], [], paths, []);
+  try {
     // Once the shape and paths are loaded, construct a validator from the loaded schema
     const db = Util.makeN3DB(loaded.data);
     const validator = Validator.construct(loaded.schema, { results: 'api' });
 
-    // Loop over each notification found and attempt to validate it against the core notification shape
-    paths.forEach(path => {
-      const result = validator.validate(db, [{ node: path, shape: Validator.start }]);
-      if (result && result.length > 0) {
-        // If the result is a success and the shape matches the data, add it to an array
-        if (result.filter(r => r.status === 'conformant').length > 0) {
-          validPaths.push(path);
-        }
+    const toValidate = paths.map(p => ({
+      node: p,
+      shape: Validator.start
+    }));
+
+    const results = validator.validate(db, toValidate);
+    results.forEach(resultItem => {
+      if (resultItem.status === 'conformant') {
+        const proofStore = new N3.Store();
+        Util.getProofGraph(resultItem.appinfo, proofStore, N3.DataFactory);
+        validQuads = [...validQuads, proofStore.getQuads()];
       }
     });
```

Comment:
It's not a biggie so no need to change it, but as a suggestion for future code: using `map` and `filter` there's no need to initialise an empty array at the top, and to keep concatenating arrays (`validQuads = [...validQuads, proofStore.getQuads()]`):

```suggestion
    results
      .filter(resultItem => resultItem.status === 'conformant')
      .map(resultItem => {
        const proofStore = new N3.Store();
        Util.getProofGraph(resultItem.appinfo, proofStore, N3.DataFactory);
        return proofStore.getQuads();
      });
```

Reply:
I can try your suggested code out. 

Originally this WAS done with .map, then .reduce. I worked with another dev (a ShEx expert) on this part of the code, but the .reduce wasn't working properly and was causing unexpected behavior.

For now, I went with a simpler, old fashioned method to get it working. I can try again with your .map and see if it works as expected now that reduce (and some of the strange internal code) has been updated and fixed.

Reply:
You could, but don't spend too much time on it. In any case it's a fine line to thread, and especially `reduce` can easily lead to code becoming _less_ easy to follow. In this case it's almost a toin coss, IMHO.